src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -851,37 +851,20 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
         // In the presence of re-exports, this is not the same as the module of the item.
         // Rather than merging all documentation into one, resolve it one attribute at a time
         // so we know which module it came from.
-        let mut attrs = item.attrs.doc_strings.iter().peekable();
-        while let Some(attr) = attrs.next() {
-            // `collapse_docs` does not have the behavior we want:
-            // we want `///` and `#[doc]` to count as the same attribute,
-            // but currently it will treat them as separate.
-            // As a workaround, combine all attributes with the same parent module into the same attribute.
-            let mut combined_docs = attr.doc.clone();
-            loop {
-                match attrs.peek() {
-                    Some(next) if next.parent_module == attr.parent_module => {
-                        combined_docs.push('\n');
-                        combined_docs.push_str(&attrs.next().unwrap().doc);
-                    }
-                    _ => break,
-                }
-            }
-            debug!("combined_docs={}", combined_docs);
+        for (parent_module, doc) in item.attrs.collapsed_doc_value_by_module_level() {
```

Comment:
This new code seems *much* less clear to me: the comments have disappeared and there's no indication on `collapsed_doc_value_by_module_level` why it would be used. Is there any reason to change this?

Reply:
Because now fragments aren't collapsed together, meaning we have to do it when we when to get them. In this case, it collapses based on the parent module.

Reply:
However adding documentation about this would be very interesting. :)

Reply:
And I just thought that actually, a little update of the function will be needed because I think it's not doing exactly what it's supposed to do (for the rendering).